src/kubernetes.cc

```diff
@@ -757,13 +761,23 @@ struct Watcher : public std::thread {
         LOG(ERROR) << name_ << " => Unlocking completion mutex";
       }
       service_->stop();
+    } else {
+      staleness_timer_->RunAsyncAfter(std::chrono::seconds(
```

Comment:
Why do you want to do this here, rather than in the staleness callback itself?

Reply:
Actually, never mind, I've misread the code. This is effectively re-scheduling the same callback for later. Ultimately, only the expiration time is changed on the underlying `boost::asio::basic_waitable_timer`, but we're not exposing that functionality in our `DelayTimer` class. Perhaps we should do that instead (e.g., add a `Snooze` method to `DelayTimer`).

Reply:
Not in this PR, I'll let Dave follow up with changing the implementation.